core/amber/src/main/scala/edu/uci/ics/texera/web/resource/UserDictionaryResource.scala

```diff
@@ -0,0 +1,238 @@
+package edu.uci.ics.texera.web.resource
+
+import edu.uci.ics.texera.web.SqlServer
+import edu.uci.ics.texera.web.model.jooq.generated.Tables.USER_DICTIONARY
+import edu.uci.ics.texera.web.model.jooq.generated.tables.daos.UserDictionaryDao
+import edu.uci.ics.texera.web.model.jooq.generated.tables.pojos.{User, UserDictionary}
+import edu.uci.ics.texera.web.resource.auth.UserResource
+import io.dropwizard.jersey.sessions.Session
+
+import javax.servlet.http.HttpSession
+import javax.ws.rs._
+import javax.ws.rs.core.{MediaType, Response}
+import scala.jdk.CollectionConverters.asScalaBuffer
+
+/**
+  * This class handles requests to read and write the user dictionary,
+  * an abstract collection of (key, value) string pairs that is unique for each user
+  * This is accomplished using a mysql table called user_dictionary.
+  * The details of user_dictionary can be found in /core/scripts/sql/texera_ddl.sql
+  */
+@Path("/users/dictionary")
+@Consumes(Array(MediaType.APPLICATION_JSON))
+@Produces(Array(MediaType.APPLICATION_JSON))
+class UserDictionaryResource {
+  final private val userDictionaryDao = new UserDictionaryDao(
+    SqlServer.createDSLContext.configuration
+  )
+
+  /**
+    * This method either retrieves all of the current in-session user's dictionary
+    * as a json object or retrieves the value corresponding to the "key" attribute of the GetRequest.
+    * @param session HttpSession
+    * @param req GetRequest (key may be null for GetDictRequest)
+    * @return Response with a UserDictionaryResponse payload, 200 ok - payload: EntryResponse or DictResponse,
+    * 401 unauthorized - payload: ErrorResponse("Invalid session"),
+    * 400 bad request - payload: ErrorResponse("Malformed JSON payload"),
+    * 422 Unprocessable Entity - payload: ErrorResponse("No such entry") (if no entry exists for provided key)
+    */
+  @POST
+  @Path("/get")
+  def getValue(@Session session: HttpSession, req: GetRequest): Response = {
+    val user = UserResource.getUser(session).orNull
+    if (user == null)
+      Response.status(Response.Status.UNAUTHORIZED).entity(ErrorResponse("Invalid session")).build()
+    else if (
+      req == null || req.requestType == 0 || (req.key == null && req.requestType == GetRequest.GET_ENTRY)
```

Comment:
instead of using the number/int, please use the enum name for this request type for readability.

Reply:
agreed